### PR TITLE
Switch ant to maven

### PR DESCRIPTION
**Pull request by mrDoctorWho**
Maven has different project structure, so I changed it as well. 

I'm not familiar with ant, but from what I've seen I suppose it's not really easy to keep both build systems at the one time, so I dropped ant and hope that's okay.

### CODE REVIEW

**Comment by pieroxy** [comment]
Thanks. I'll have a look this WE.

**Comment by pieroxy** [comment]
So, a few questions and remarks:

1. First of all thanks for the pull request
2. I will not merge this as it is because I have a small refactoring going on (and this code is generated). But this tells me the exact changes I have to make, so it is great.
3. I expect to have this done next week or at worst the week after that.

File: cleanupJavadoc.sh [hunk 1]
```diff
@@ -1,2 +0,0 @@
-find release/doc -name "*.html" -exec sed -i "s/.*Generated by javadoc.*//" {} \;
```
Comment (pieroxy) [inline cleanupJavadoc.sh]: Why did you remove this file? I added it to cleanup the javadoc so that generating the JavaDoc doesn't result is a file change if the corresponding Java file did not change.

File: build.xml [hunk 2]
```diff
@@ -1,45 +0,0 @@
-<project name="user-agent-tools" default="compile" basedir=".">
```
Comment (pieroxy) [inline build.xml]: Why did you remolve this file? I'd rather use ant for my personal workflow, can't both coexist ?

File: test.sh [hunk 3]
```diff
@@ -1 +1 @@
-java -cp build/user-agent-detector.jar net.pieroxy.ua.tooling.UserAgentTester test test-data/database.gz
+java -cp target/user_agent_detector-2015-10-31-SNAPSHOT.jar net.pieroxy.ua.tooling.UserAgentTester test src/test/resources/database.gz
```
Comment (pieroxy) [inline test.sh]: What does the timestamp mean there? Will this be stable over time ?

Sorry about the stupid questions, but I've managed my way through the Java world without approaching maven so I'm quite the n00b when it comes to that.

Comment (mrDoctorWho) [inline test.sh]: I just copied your version string. Actually using test.sh is not good, you might want to replace the .sh script by src/java/test/sometestclass.java and then maven will run tests on deploy/compile.

File: build.xml (hunk 2 above)
Comment (mrDoctorWho) [inline build.xml]: Well, it's up to you. As you don't know maven, I don't know ant.

If ant can deal with the new project structure, then you can keep both. Actually, maven could do that but I considered that ant is not needed anymore. I understand if you reject my changes.

File: cleanupJavadoc.sh (hunk 1 above)
Comment (mrDoctorWho) [inline cleanupJavadoc.sh]: Sorry, probably I missed some point where the docs were generated or something like that.

File: test.sh (hunk 3 above)
Comment (pieroxy) [inline test.sh]: Right, thanks.

File: build.xml (hunk 2 above)
Comment (pieroxy) [inline build.xml]: Allright, I'll keep ant.

File: cleanupJavadoc.sh (hunk 1 above)
Comment (pieroxy) [inline cleanupJavadoc.sh]: Actually I'm wondering if it is a good idea to keep the generated javadoc in the GIT repo. I like the idea of checking out and having the doc ready on your hard drive, but on the other hand it is generated stuff, so it has little value.

Comment (mrDoctorWho) [inline cleanupJavadoc.sh]: No, it's not good, as well as keeping binary packages. Actually, there's a javadoc plugin for maven.

Comment (hth) [inline cleanupJavadoc.sh]: @pieroxy Agree with @mrDoctorWho on generated doc. Generated doc will become a nuisance to check in every time a change is made.

**Comment by pieroxy** [comment]
Allright, so I pushed a commit that supposedly cover this pull request. Can you give it a try and let me know if all is covered?

If so, I'll close this one.

Thanks for the help.

**Comment by mrDoctorWho** [comment]
@pieroxy sorry for a long silence, wasn't able to test it before.

So, I tested it, works like a charm. But maybe you should consider doing auto-versioning? Maven supports that.

**Comment by pieroxy** [comment]
Yes, I'm in the process of adding the project to maven central, so all build stuff will be refactored in the coming days.

Do you have an example of a GitHub project that does auto-versioning so I can have a look?

**Comment by mrDoctorWho** [comment]
@pieroxy unfortunately, I don't have an example, but I think you should start here: http://maven.apache.org/maven-release/maven-release-plugin/

**Comment by pieroxy** [comment]
Thanks. I will.
